src/main/java/org/openhab/binding/zwave/handler/ZWaveControllerHandler.java

```diff
@@ -135,17 +137,23 @@ public void initialize() {
             networkKey = (String) param;
         }
         if (networkKey.length() == 0) {
-            logger.debug("No network key set by user - using random value.");
-
-            // Create random network key
-            networkKey = "";
-            for (int cnt = 0; cnt < 16; cnt++) {
-                int value = (int) Math.floor((Math.random() * 255));
-                if (cnt != 0) {
-                    networkKey += " ";
-                }
-                networkKey += String.format("%02X", value);
+            logger.debug("No network key set by user - using secure random value.");
+
+            // Create network key
+            byte[] networkKeyBytes = new byte[16];
+            try {
+                SecureRandom.getInstanceStrong().nextBytes(networkKeyBytes);
+            } catch (NoSuchAlgorithmException e) {
+                logger.warn("SecureRandom strong init failed - falling back to default SecureRandom.", e);
```

Comment:
I would suggest to make this a debug log - I think WARN is too strong and is not within the guidelines.

Reply:
Will change this.  I need to brush up on the guidelines for logging.  Have a lot of statements in my S2 branch